sycl/test-e2e/bindless_images/mipmap/mipmap_read_3D.cpp

```diff
@@ -10,29 +10,39 @@
 // Uncomment to print additional test information
 // #define VERBOSE_PRINT
 
-class image_addition;
-
-int main() {
+template <typename DType, sycl::image_channel_type CType, typename KernelName>
```

Comment:
Instead of a `KernelName` argument, you could just define a class above that is templated on `DType` and `CType` and use that in here. It should still be pretty readable if they fail.